File: server/plugin/merchant/router/merchant.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

var Merchant = new(merchant)

type merchant struct {}

// Init 初始化 商户信息 路由信息
func (r *merchant) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
	    group := private.Group("merchant").Use(middleware.OperationRecord())
		group.POST("createMerchant", apiMerchant.CreateMerchant)   // 新建商户信息
		group.DELETE("deleteMerchant", apiMerchant.DeleteMerchant) // 删除商户信息
		group.DELETE("deleteMerchantByIds", apiMerchant.DeleteMerchantByIds) // 批量删除商户信息
		group.PUT("updateMerchant", apiMerchant.UpdateMerchant)    // 更新商户信息
	}
	{
	    group := private.Group("merchant")
		group.GET("findMerchant", apiMerchant.FindMerchant)        // 根据ID获取商户信息
		group.GET("getMerchantList", apiMerchant.GetMerchantList)  // 获取商户信息列表
	}
	{
	    group := public.Group("merchant")
	    group.GET("getMerchantPublic", apiMerchant.GetMerchantPublic)  // 商户信息开放接口
	}
}

File: web/src/plugin/merchant/api/merchant.js
import service from '@/utils/request'
// @Tags Merchant
// @Summary 创建商户信息
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.Merchant true "创建商户信息"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /merchant/createMerchant [post]
export const createMerchant = (data) => {
  return service({
    url: '/merchant/createMerchant',
    method: 'post',
    data
  })
}

// @Tags Merchant
// @Summary 删除商户信息
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.Merchant true "删除商户信息"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /merchant/deleteMerchant [delete]
export const deleteMerchant = (params) => {
  return service({
    url: '/merchant/deleteMerchant',
    method: 'delete',
    params
  })
}

// @Tags Merchant
// @Summary 批量删除商户信息
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除商户信息"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /merchant/deleteMerchant [delete]
export const deleteMerchantByIds = (params) => {
  return service({
    url: '/merchant/deleteMerchantByIds',
    method: 'delete',
    params
  })
}

// @Tags Merchant
// @Summary 更新商户信息
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.Merchant true "更新商户信息"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /merchant/updateMerchant [put]
export const updateMerchant = (data) => {
  return service({
    url: '/merchant/updateMerchant',
    method: 'put',
    data
  })
}

// @Tags Merchant
// @Summary 用id查询商户信息
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query model.Merchant true "用id查询商户信息"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /merchant/findMerchant [get]
export const findMerchant = (params) => {
  return service({
    url: '/merchant/findMerchant',
    method: 'get',
    params
  })
}

// @Tags Merchant
// @Summary 分页获取商户信息列表
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取商户信息列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /merchant/getMerchantList [post]
export const getMerchantList = (data) => {
  return service({
    url: '/merchant/getMerchantList',
    method: 'post',
    data
  })
}
// @Tags Merchant
// @Summary 不需要鉴权的商户信息接口
// @Accept application/json
// @Produce application/json
// @Param data query request.MerchantSearch true "分页获取商户信息列表"
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /merchant/getMerchantPublic [get]
export const getMerchantPublic = () => {
  return service({
    url: '/merchant/getMerchantPublic',
    method: 'get',
  })
}
